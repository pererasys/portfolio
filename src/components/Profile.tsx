/**
 * @author Andrew Perera
 * Copyright (C) 2021
 */

import Image from "next/image";
import Link from "next/link";

import styles from "../styles/Profile.module.scss";

const Profile = () => {
  return (
    <section className={styles.root}>
      <div className={styles.column}>
        <h1>About Me</h1>
      </div>
      <div className={styles.column}>
        <Image
          src="/me.jpeg"
          height={300}
          width={300}
          layout="intrinsic"
          className={styles.image}
        />
        <p className={styles.content}>
          I'm Andrew! I recently graduated from the University of South Carolina
          with a major in Computer Science and minor in Business. I started
          teaching myself about software engineering two years ago and am
          currently a developer at Vanguard.
        </p>
        <Link href="/about">
          <a>See more</a>
        </Link>
      </div>
    </section>
  );
};

export default Profile;
