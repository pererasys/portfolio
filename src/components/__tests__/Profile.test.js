import { create } from "react-test-renderer";

import Profile from "../Profile";

describe("<Profile />", () => {
  const setup = () => {
    const component = create(<Profile />);

    return {
      component,
    };
  };

  it("should render without crashing", () => {
    const { component } = setup();

    expect(component.toJSON()).toMatchSnapshot();
  });
});
