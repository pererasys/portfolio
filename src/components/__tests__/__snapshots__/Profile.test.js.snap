// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should render without crashing 1`] = `
<section>
  <div>
    <h1>
      About Me
    </h1>
  </div>
  <div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <noscript>
        <img
          decoding="async"
          src="/me.jpeg"
          srcSet="/_next/image?url=%2Fme.jpeg&w=384&q=75 1x, /_next/image?url=%2Fme.jpeg&w=640&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
      <img
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </div>
    <p>
      I'm Andrew! I recently graduated from the University of South Carolina with a major in Computer Science and minor in Business. I started teaching myself about software engineering two years ago and am currently a developer at Vanguard.
    </p>
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      See more
    </a>
  </div>
</section>
`;
